bigframes/core/compile/compiler.py

```diff
@@ -88,6 +95,80 @@ def compile_readlocal(node: nodes.ReadLocalNode, ordered: bool = True):
         return ordered_ir.to_unordered()
 
 
+@_compile_node.register
+def compile_readtable(node: nodes.ReadTableNode, ordered: bool = True):
+    full_table_name = f"{node.project_id}.{node.dataset_id}.{node.table_id}"
+    used_columns = (
+        *node.schema.names,
+        *[i for i in node.primary_key if i not in node.schema.names],
+    )
+    # Physical schema might include unused columns, unsupported datatypes like JSON
+    physical_schema = ibis.backends.bigquery.BigQuerySchema.to_ibis(
+        list(i for i in node.physical_schema if i.name in used_columns)
+    )
+    if node.snapshot_time is not None or node.sql_predicate is not None:
+        import bigframes.session._io.bigquery
+
+        sql = bigframes.session._io.bigquery.to_query(
+            full_table_name,
+            columns=used_columns,
+            sql_predicate=node.sql_predicate,
+            time_travel_timestamp=node.snapshot_time,
+            # These parameters should not be used
+            index_cols=(),
+            max_results=None,
+        )
+        ibis_table = ibis.backends.bigquery.Backend().sql(
+            schema=physical_schema, query=sql
+        )
+    else:
+        ibis_table = ibis.table(physical_schema, full_table_name)
+
+    if ordered:
+        if node.primary_key:
+            ordering_value_columns = tuple(
+                bf_ordering.ascending_over(col) for col in node.primary_key
+            )
+            if node.primary_key_sequential:
+                integer_encoding = bf_ordering.IntegerEncoding(
+                    is_encoded=True, is_sequential=True
+                )
+            else:
+                integer_encoding = bf_ordering.IntegerEncoding()
+            ordering = bf_ordering.ExpressionOrdering(
+                ordering_value_columns,
+                integer_encoding=integer_encoding,
+                total_ordering_columns=frozenset(node.primary_key),
+            )
+            hidden_columns = ()
+        else:
```

Comment:
This is getting a bit long for me to track indentation-wise. Could we make some helper functions, please?